### DISCUSSION TITLE
Talos 1.4.6 and 1.5.0-alpha.1 won't load on bare metal EX101 in Hetzner

### DISCUSSION THREAD

**Discussion by eshikhov** [siderolabs/talos#7465] [Q&A] [answered]
Hi!

I have a bare metal EX101 server at Hetzner. The configuration added memory up to 128GB and another ssd disk 980GB for talos. I'm trying to run talos on it. After the boot menu appears and the first item is selected, nothing happens, only a black screen. I tried both booting from an iso image and unpacking the image to a hard drive. The result is the same. I took version 1.4.6 and 1.5.0-alpha.1
Please tell me how you can debug talos, how to find out what the download is hanging on. Maybe there are some keys for the kernel to debug talos?

**Comment by smira** [comment]
It might be as simple as wrong console settings. I assume you're using a video console, and that is sometimes tricky.

You can try `console=tty1` and other Linux kernel console-related settings.

If you have access to a serial console, that works way better.

You can check that Talos listens on port :50000 after boot, in that case it works despite the fact that you don't have console access.

**Comment by eshikhov** [comment]
I see the output in the console if you enable debug log. 

![Снимок экрана 2023-07-12 в 13 43 22](https://github.com/siderolabs/talos/assets/8626973/e5afa53e-13b2-4c79-86db-bddc56908d48)

this is the whole output, the download does not go further

**Reply by smira** [reply]
This is expected as EFI stub prints using UEFI, but Linux will switch to using its own methods to print to the console.

**Comment by eshikhov** [comment]
there is no option to write the boot log to a file?

**Reply by smira** [reply]
There is no filesystem during the early boot, where would it write it? Talos runs unmodified Linux kernel, so any generic Linux options work.
I would rather ask for a serial console from Hetzner, this is the easiest way.

**Comment by eshikhov** [comment]
In Hetzner they answered that they can only give KVM, are there any other options how to debug?

**Reply by smira** [reply]
Please see my first answer https://github.com/siderolabs/talos/discussions/7465#discussioncomment-6425811

**Reply by frezbo** [reply]
you shoulod try with different console options. Eg:

* `console=ttyS0`
* `console=ttyAMA0`
* `console=tty0` there is not standard for this , so it's kind of a trial and error process unless the motherboard/hardware manufactures provides this info (some actually do)

**Comment by eshikhov** [comment, marked as answer]
The problem was solved by removing the parameters from the kernel
earlyprintk=ttyS0 console=ttyS0 console=tty0
consoleblank=0
printk.devkmsg=on now you can see the installation process and the console itself in KVM, also added the "ip=" parameter to the kernel for the network to work correctly.

thanks for the ideas

**Reply by vladimirfx** [reply]
When we try to upgrade nodes with removed `console` args it stops booting again. We added necessary `ip=` args to `machine.install.extraKernelArgs` but this does not remove any default args which break boot.

@smira How to remove some default kernel args or override them? May be some special syntax in `machine.install.extraKernelArgs`?

**Reply by smira** [reply]
good question, I don't think Talos supports that at the moment. I'm curious, which argument breaks the boot? just trying to figure out more of a long-term solution.

**Reply by frezbo** [reply]
it's weird that removing console args makes it works, adding wrong console args should have no effect on boot

**Reply by vladimirfx** [reply]
Just one default kernel arg broke our boot - `console=ttyS0` . Tested on 1.5.0-beta.0 . 
Seems it is specifics of our hardware so no need to remove this param in default conf.
Please provide the way to remove default kernel. Removing by exact match is enough.

**Reply by smira** [reply]
this is really interesting, as `ttyS0` and `tty0` should work independent of each other... :shrug:

**Reply by frezbo** [reply]
hmm I found this, https://github.com/systemd/systemd/issues/15656 but no idea why this does happen

**Reply by frezbo** [reply]
hmm i think the kernel is missing driver for the serial console for the particular hardware and hangs trying to output there

**Reply by smira** [reply]
hm.... but we could simply swap the args? this is my understanding from reading this...

@vladimirfx does it help if you edit the args to be `console=ttyS0 console=tty0` (with `ttyS0` first?)

**Reply by frezbo** [reply]
yeh, swapping might work

**Reply by vladimirfx** [reply]
I don't think so. According to kernel documentation output is directed to all outputs enumerated by console arg. Last one is used for /dev/console. So if kernel can't find driver for serial console boot is hang regardless of order. 

Exactly in this order arguments are now (by default). 

I will check opposite variant tomorrow.

**Reply by vladimirfx** [reply]
https://www.kernel.org/doc/html/v4.12/admin-guide/serial-console.html

**Reply by vladimirfx** [reply]
I've tried to change console arguments order - didn't help. 

Can I file issue for remove arg feature?

UPDATE: checked on 1.5.8-beta.1

**Reply by smira** [reply]
> Can I file issue for remove arg feature?

yes, please

**Comment by vladimirfx** [comment]
> hm.... but we could simply swap the args? this is my understanding from reading this...
> 
> @vladimirfx does it help if you edit the args to be `console=ttyS0 console=tty0` (with `ttyS0` first?)

It's current default order.

**Reply by frezbo** [reply]
@vladimirfx could you check if you remove all `console=` args, talos boots fine and console logs are viewable?

**Reply by vladimirfx** [reply]
It is what we done fist thanks to https://github.com/siderolabs/talos/discussions/7465#discussioncomment-6439832 . Yes it woks. Later I've track down that only `console=ttyS0` is problematic.

**Reply by frezbo** [reply]
ahh perfect, we're planning to drop all console args for metal platform as kernel does the automatic handling, so we don;t need to add a feature to remove args which is cumbersome

**Comment by Lokeshvasudevan** [comment]
Hi Team,

Im also facing the same issue with the 1.12.5. I have tried the workaround suggested here. However im still getting the back screen. After selecting the option.

Talos (v1.12.5)

Please suggest.
